fix test_qgsserver_wms_getmap to work with qt6

benoitdm-oslandia opened the pull request:
* update result masks and some xml tags to work with qt6
* reactivate test in qt6

Funded By Oslandia

benoitdm-oslandia committed:
fix(test_qgsserver_wms_getmap): update masks and xml tags to work with qt6 and reactivate test

nyalldawson commented on tests/src/python/test_qgsserver_wms_getmap.py:
```diff
@@ -1561,7 +1561,7 @@ def test_wms_getmap_filter_ogc_v2(self):
         # without namespace (only with prefix)
         filter = (
-            "<fes:Filter>"
+            "<fes:Filter xmlns:fes='http://www.opengis.org/fes'>"
```
The comment above explicitly says "without namespace". Isn't this breaking what the test is designed to check?

benoitdm-oslandia replied:
Indeed, I missed the comment! Without a namespace, it makes the XML not valid so I do not understand what is expected here or how it should work.

I investigated further and the error throws by the test is from https://github.com/benoitdm-oslandia/QGIS/blob/79bb4814dfd689d60eeaf15f28455e0321c27b1c/src/server/services/wms/qgswmsrenderer.cpp#L3563. What I do not understand (again) is how this can works with Qt6 as the `setContent` function has not the same parameters as Qt5?! Can you enlighten me?

benoitdm-oslandia replied:
Ok it is now a deprecated function and hidden in the documentation. I will fix

nyalldawson replied:
This shouldn't have been marked resolved -- it's still outstanding

benoitdm-oslandia replied:
wrong click!

benoitdm-oslandia replied:
I have made a (ugly) fix

troopa81 replied:
As stated [here](https://doc.qt.io/qt-6/xml-changes-qt6.html#qdom-and-qdomdocument)

> Undeclared namespace prefixes are no longer allowed.

That's why it failed with Qt6. I think a way to nicely fix these would be to use the [setContent](https://doc.qt.io/qt-6/qdomdocument.html#setContent-2) that takes a QXmlStreamReader and call [addExtraNamespaceDeclaration](https://doc.qt.io/qt-6/qxmlstreamreader.html#addExtraNamespaceDeclaration) on it before so we are sure its gonna work when parsing the filter

nyalldawson replied:
That approach sounds reasonable to me!

benoitdm-oslandia committed:
fix(qgswmsrenderer): missing namespace in filter for Qt6

Add a trick to handle filter without namespace as Qt6 is less tolerant than Qt5.
